src/frontend/screens/PhotosModal/index.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useEffect, useMemo, useRef, useState } from "react";
+import { Dimensions, View, StyleSheet } from "react-native";
+import { NavigationState, TabView, Route } from "react-native-tab-view";
+
+import IconButton from "../../sharedComponents/IconButton";
+import Button from "../../sharedComponents/Button";
+import { CloseIcon, DeleteIcon } from "../../sharedComponents/icons";
+import { filterPhotosFromAttachments } from "../../lib/utils";
+import { useObservation } from "../../hooks/useObservation";
+import { useDraftObservation } from "../../hooks/useDraftObservation";
+import PhotoView from "../../sharedComponents/PhotoView";
+import api from "../../api";
+import type { NavigationProp } from "../../types";
+import { useBottomSheetModal } from "../../sharedComponents/BottomSheetModal";
+import { ConfirmDeleteModal } from "./ConfirmDeleteModal";
+import Text from "../../sharedComponents/Text";
+import { defineMessages, FormattedMessage } from "react-intl";
+import { TouchableNativeFeedback } from "../../sharedComponents/Touchables/index.ios";
+
+interface customRoute extends Route {
+  uri?: string;
+  error?: boolean;
+  capturing?: boolean;
+}
+
+const m = defineMessages({
+  deleteImage: {
+    id: "screens.PhotosModal",
+    defaultMessage: "Delete Image",
+  },
+});
+
+const PhotosModal = ({ navigation }: { navigation: NavigationProp }) => {
+  const observationId = navigation.getParam("observationId");
+  const [index, setIndex] = useState(navigation.getParam("photoIndex") || 0);
+  const [{ observation }] = useObservation(observationId);
+  const [{ photos: draftPhotos }] = useDraftObservation();
+  const [showHeader, setShowHeader] = useState(true);
+  const { sheetRef, openSheet, closeSheet, isOpen } = useBottomSheetModal({
+    openOnMount: false,
+  });
+
+  function toggleShowHeader() {
```

Comment:
think i'd fix this up a bit by doing something along the lines of:

```ts
  const { sheetRef, openSheet, closeSheet, isOpen } = useBottomSheetModal({
    openOnMount: false,
  });

  function toggleShowHeader() {
    setShowHeader(previous => {
      if (isOpened) return false;
      return !previous;
    });
  }

  useEffect(() => {
    if (isOpened) {
      setShowHeader(false);
    }
  }, [isOpened]);
```

where `isOpen` from the hook is React state and not a ref

Reply:
This seems like a stylisic preference and the way I have it coded feels easier to read due to the explicit conditions that I am pointing out in my if statements.

I do see the utility of the useeffect here, but due to the simple mechanics it does seem like overkill